test/suites/ipv6/suite_test.go

```diff
@@ -94,4 +100,58 @@ var _ = Describe("IPv6", func() {
 		})
 		Expect(internalIPv6Addrs).To(HaveLen(1))
 	})
+	It("should provision a static IPv6 prefix with node launch and set IPv6 as primary in the primary network interface", func() {
+		clusterDNSAddr := env.ExpectIPv6ClusterDNS()
+		nodeClass.Spec.Kubelet = &v1.KubeletConfiguration{ClusterDNS: []string{clusterDNSAddr}}
+		Expect(disableVPCCNIProvisioning(true)).ToNot(HaveOccurred())
+		pod := coretest.Pod()
+		env.ExpectCreated(pod, nodeClass, nodePool)
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+		output, err := env.EC2API.DescribeInstances(env.Context, &ec2.DescribeInstancesInput{
+			InstanceIds: []string{coretest.Pod().Spec.NodeName},
+		})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(output.Reservations).To(BeNumerically(">=", 1))
+		Expect(len(output.Reservations[0].Instances)).To(BeNumerically(">=", 1))
+		Expect(len(output.Reservations[0].Instances[0].NetworkInterfaces)).To(BeNumerically(">=", 1))
+		Expect(output.Reservations[0].Instances[0].NetworkInterfaces[0].Ipv6Prefixes).To(HaveLen(1))
+		_, hasIPv6Primary := lo.Find(output.Reservations[0].Instances[0].NetworkInterfaces[0].Ipv6Addresses, func(ip types.InstanceIpv6Address) bool {
+			return lo.FromPtr(ip.IsPrimaryIpv6)
+		})
+		Expect(hasIPv6Primary).To(BeTrue())
+		Expect(disableVPCCNIProvisioning(false)).ToNot(HaveOccurred())
+	})
 })
+
+// disable VPC CNI provisioning on network interfaces and IPs
+func disableVPCCNIProvisioning(disable bool) error {
```

Comment:
Juice may not be worth the squeeze here to disable this -- checking primary seems fine as long as we have the functional testing around what we are setting in the launch template parameters

Reply:
Disabling VPC CNI provisioning can guarantee the prefix comes from LT. Sounds like a validate check for me. Let me know if you refer not to add complexity in the test.